### DISCUSSION TITLE
Codegen: exclude unnecessary type classes

### DISCUSSION THREAD

**Discussion by Gavrilov-Ivan** [Netflix/dgs-framework#908] [Ideas]
Hello

Currently dgs-codegen has settings parameter `generateDataTypes` with 'all-or-nothing' logic: generate classes for all types in schema or not generate at all. 
If you have a big schema with huge amount of types but include a few queries only you will get classes for a whole scope of types even if included queries don't use them.

It will be nice to have an ability to generate only the types required according to `includeQueries`/`includeMutations`/`includeSubscriptions` as it already done for input types (the same way or with additional settings parameter).

**Comment by srinivasankavitha** [comment]
Thanks for the feature request. We can look into adding that in the future.
